BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -302,9 +321,15 @@ - (void)executeOperation
 
         if (self.error == nil && ![self isCancelled])
         {
-            self.connection = [[NSURLConnection alloc] initWithRequest:self.APIRequest delegate:self];
-            BOXLog(@"Starting %@", self);
-            [self startURLConnection];
+            //FIXME: remove shouldUseSessionTask and this condition after switching
+            //completely from NSURLConnection to NSURLSessionTask
+            if (self.shouldUseSessionTask == YES && self.sessionTask != nil) {
```

Comment:
you're right, we should just rely on self.shouldUseSessionTask == YES while we transition to nsurlsession gradually.